Debian package fix lintian errors

yurtpage opened the pull request:
@kvvloten please review.

The [set cmake_minimum_required to 3.5](https://github.com/altlinux/admc/commit/802f1a9781fc8e117cc066b731ef810fcd84169f) I made to fix a deprecation warning.

Now we have two ways to build a deb package: the standard `debuild` and the `make package` to build the deb package directly from the cmake with CPack. I found this confusing and unless the CPack is really used I would propose to remove it.

*UPD fixed in 468dcc8* Another problem is that tests are included by default unless the `ADMC_BUILD_DEB` var is not set. Because of this we have to specify the `ADMC_BUILD_DEB` in the `debian/rules` file.
Maybe the test should be implemented with the CMake add_test or something like that to make it more standard. The the debhelper will just skip the tests in the convenient way.

yurtpage committed:
README.md: fix formatting

Use header level 2 for sections.
Specify shell as language for commands.

yurtpage committed:
.gitignore: ignore debian generated files

yurtpage committed:
.gitignore: ignore generated object files

yurtpage committed:
CMakeLists.txt: set cmake_minimum_required to 3.5

yurtpage committed:
debian: set email for debian package maintainer

yurtpage committed:
debian: use ${shlibs:Depends} to automatically generate the list of dep libs

Two libraries are added manually because they aren't detected by shlibs

yurtpage committed:
debian/copyright: fix

yurtpage committed:
debian: use debhelper-compat level 11 instead of compat file

Also replace deprecated pkg-config with pkgconf

yurtpage committed:
debian: add missing source format

yurtpage committed:
debian: control: set section to admin

yurtpage committed:
README.md: add the description and move the screenshot to the top

yurtpage committed:
debian/changelog: add a new release

yurtpage committed:
debian/rules: remove --buildsystem=cmake since we updated compat

yurtpage committed:
debian: control: extend description

august-alt commented on debian/rules:
```diff
@@ -4,7 +4,7 @@ export DH_VERBOSE=1
 
 %:
-	dh $@ --buildsystem=cmake
+	dh $@
```
Here we was selecting cmake manually instead of automatically detecting it. Can you clarify why this line was changed?

yurtpage committed:
debian/control: add Vcs-Git

yurtpage committed:
CMakeLists.txt: Use BUILD_TESTING instead of ADMC_BUILD_DEB

The CMake has a built-in variable BUILD_TESTING (default ON) to compile tests.

When building a debian package, the debhelper configures the CMake with this option OFF.
So now we don't need to pass the DADMC_BUILD_DEB=ON and we can simplify the debian/rules

yurtpage committed:
debian/control: add Bugs

august-alt commented on README.md:
What is the purpose of modifying the README.md file in a pull request that specifically addresses Debian packaging? The changes to README.md are unrelated to the purpose of this pull request, please consider removing them and submitting a separate pull request for those changes if you still wish to update the file."

august-alt commented on CMakeLists.txt:
```diff
@@ -52,9 +52,9 @@ include(ClangFormat)
 
 add_subdirectory(src)
-if(NOT ADMC_BUILD_DEB)
+if(BUILD_TESTING)
```
If I recall correctly, the `BUILD_TESTING` variable is set up by CTest, which is not currently enabled in this project (see
https://cmake.org/cmake/help/latest/module/CTest.html
). Instead, we use the `ADMC_BUILD_DEB` variable, which is `OFF` by default. We need to consider one of three options: creating and setting a separate variable to explicitly enable or disable testing, setting `ADMC_BUILD_DEB` to `ON` by default, or completely refactoring the test setup.

yurtpage committed:
Merge branch 'master' into deb
